lib/spack/spack/binary_distribution.py

```diff
@@ -735,22 +731,93 @@ def generate_package_index(cache_prefix):
 
     tty.debug('Retrieving spec.yaml files from {0} to build index'.format(
         cache_prefix))
+
+    all_mirror_specs = {}
+
     for file_path in file_list:
         try:
             yaml_url = url_util.join(cache_prefix, file_path)
             tty.debug('fetching {0}'.format(yaml_url))
             _, _, yaml_file = web_util.read_from_url(yaml_url)
             yaml_contents = codecs.getreader('utf-8')(yaml_file).read()
-            # yaml_obj = syaml.load(yaml_contents)
-            # s = Spec.from_yaml(yaml_obj)
+            spec_dict = syaml.load(yaml_contents)
             s = Spec.from_yaml(yaml_contents)
-            db.add(s, None)
-            db.mark(s, 'in_buildcache', True)
+            num_deps = 0
+            for d in s.traverse(root=False):
+                num_deps += 1
```

Comment:
What about
```suggestion
            num_deps = len(list(s.traverse(root=False)))
```
or just do that inline when constructing the dict